events/worker.go

```diff
@@ -10,7 +10,12 @@ type EventLocker func(event *Event) locks.Locker
 
 func nopLocker(_ *Event) locks.Locker { return locks.NopLocker() }
 
-func resourceIDLocker(event *Event) locks.Locker { return locks.KeyLocker(event.ResourceID) }
+func resourceIDLocker(event *Event) locks.Locker {
+	if event.ResourceID == "" {
+		return locks.NopLocker()
+	}
+	return locks.KeyLocker(event.ResourceID)
```

Comment:
I think in py-agent and go agent is that go-agent locks events in the framework which is eventsubcriber, and py-agent locks the events by locking instances in methods like do_instance_activate. Seem like py-agent didn't use locks in do_instance_inspect. Not sure if I understand that correctly.